File: settlers-of-banambitha/src/app/service/shop/shop.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { environment } from 'src/environments/environment';
import { UserService } from '../user/user.service';
import { environment } from '../../../environments/environment.prod';

@Injectable({
  providedIn: 'root'
})
export class ShopService {

  private static readonly baseUrl     = environment.baseUrl + "/producto"
  private static readonly adquiridos  = ShopService.baseUrl + "/adquiridos"
  private static readonly adquirir    = ShopService.baseUrl + "/adquirir"
  private static readonly disponibles = ShopService.baseUrl + "/disponibles"
  private static readonly httpOptions = {
    headers: new HttpHeaders({ 'Content-Type': 'application/json' }),
    withCredentials: true,
    observe: 'response' as 'response'
  }

  constructor( private http: HttpClient, private userService: UserService ) { }

  public ObtenerProductosDisponibles(){
    return this.http.get(ShopService.disponibles + "/" + this.userService.getUsername(), ShopService.httpOptions)
  }

  public ObtenerProductosAdquiridos(){
    return this.http.get(ShopService.adquiridos + "/" + this.userService.getUsername(), ShopService.httpOptions)
  }

  public async adquirproducto(id_producto: String){
    let msg = {
      usuario_id: this.userService.getUsername(),
      producto_id: id_producto
    }
    let response = await this.http.put(ShopService.adquirir, msg, ShopService.httpOptions).toPromise()
    this.userService.updateUserData(response.body["usuario"])
  }
}

File: desktopCatan/src/main/java/es/susangames/catan/service/ShopService.java
package es.susangames.catan.service;

import org.json.*;
import java.io.IOException;

//http://localhost:8080
//https://catan-backend-app.herokuapp.com/
public class ShopService {
    private static HttpService netService;
    //private static final String baseUrl = "http://localhost:8080/producto";
    private static final String baseUrl = "https://catan-backend-app.herokuapp.com/producto";
    private static final String adquiridosUrl = baseUrl + "/adquiridos";
    private static final String adquirirUrl = baseUrl + "/adquirir";
    private static final String disponiblesUrl = baseUrl + "/disponibles";

    public ShopService() {
        netService = new HttpService();
    }

   public static JSONArray obtenerProductosDisponibles() {
    JSONArray response;
        try {
            response = netService.getArr(disponiblesUrl + "/" + UserService.getUsername());
        } catch(Exception e) {
            return null;
        }
        return response;
    }

    public static void adquirirProducto(String id_producto) {
        JSONObject myObject = new JSONObject();
        myObject.put("usuario_id", UserService.getUsername());
        myObject.put("producto_id", id_producto);
        String response;
        try {
            response = netService.put(adquirirUrl, myObject.toString());
            JSONObject arr = new JSONObject(response);
            JSONObject aux = new JSONObject(arr.get("usuario").toString());
            UserService.fillData(aux);
        } catch(Exception e) {}
        
    }


    public static JSONArray obtenerProductosAdquiridos() {
        JSONArray response;
        try {
            response = netService.getArr(adquiridosUrl + "/" + UserService.getUsername());
        } catch(Exception e) {
            return null;
        }
        return response;
    }

}